### PR TITLE
ALC-143 mistX Quotes

### PR DESCRIPTION
**Pull request by Dire-0x**
### Refactor `useMinerBribe` hook
The miner bribe hook is no longer needed since we are using benchmarked gas amounts and calculating the miner bribe in the SDK. In replacement, we have a new hook `useLatestGasPrice` that fetches the latest block on blockNumber changes and returns the gas price of the last transaction on the block.

### Uses the refactored SDK
In order to run this branch you will need to checkout the `alc-189-refactor` SDK branch and run `yarn build` locally. After its built run `yarn link`. From the FE directory checkout this branch `alc-143-mistx-router-quote` and then run `yarn link @alchemistcoin/sdk`. This will tell yarn to use the locally built version of the SDK.

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -212,7 +212,7 @@ export const INITIAL_ALLOWED_SLIPPAGE = 50
 // 20 minutes, denominated in seconds
 export const DEFAULT_DEADLINE_FROM_NOW = 60 * 20
 // default bribe margin, in bips
-export const INITIAL_BRIBE_MARGIN = 5
+export const INITIAL_BRIBE_MARGIN = 120
```
Comment (juice0x) [inline src/constants/index.ts]: Do we want to leave miner bribe at 120 for now?

File: src/state/swap/hooks.ts [hunk 2]
```diff
@@ -164,22 +167,30 @@ export function useDerivedSwapInfo(): {
...
     Exchange.UNI,
     inputCurrency ?? undefined,
-    !isExactIn ? parsedAmount : undefined
+    !isExactIn ? parsedAmount : undefined,
+    gasPriceToBeat,
+    BigNumber.from(userBribeMargin)
   )
   const bestTradeExactOutSushi = useTradeExactOut(
-    Exchange.SUSHI,
+    Exchange.UNI,
```
Comment (juice0x) [inline src/state/swap/hooks.ts:189]: May be able to use bestTradeExactOut solely since bestTradeExactOutSushi is the same

**Review by juice0x — APPROVED** [review]

File: src/constants/index.ts (hunk 1 above)
Comment (Dire-0x) [reply src/constants/index.ts]: Yea good point, i'll lower this number to 25 for now. We can address this further in https://linear.app/alchemistcoin/issue/ALC-318/advanced-swap-settings

File: src/state/swap/hooks.ts (hunk 2 above)
Comment (Dire-0x) [reply src/state/swap/hooks.ts:189]: We can leave this for now. @swgz9 needs to take a deeper dive into why the sushi exchange trade calculations are not working properly. We just set the sushi trade calculations to use UNI as a quick patch.

**Pull request merged**